ts/defs.bzl

```diff
@@ -167,6 +173,12 @@ def ts_project(
 
             See [docs/transpiler.md](/docs/transpiler.md) for more details.
 
+        declaration_transpiler: A custom transpiler tool to run that produces the TypeScript declaration outputs instead of `tsc`.
```

Comment:
TODO: names, should be similar to the existing `transpiler` attribute:
* declaration_transpiler
* dts_transpiler
* types_transpiler
* ?

Reply:
Decided on `declaration_transpiler`